### PR TITLE
Extend ImportGroup states

### PR DESCRIPTION
**Pull request by czbarth**
The possible states of import groups are extended to be able to cover more scenarios.

### CODE REVIEW

File: src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs [hunk 1]
```diff
@@ -27,11 +27,12 @@ public static class ImportGroupExtensions
     /// Creates an <see cref="ImportGroup"/> that is complete and can be imported at once.
     /// </summary>
     /// <param name="associatedFile">The file to be imported.</param>
-    public static ImportGroup CreateCompleteImportGroup(this IFileSource associatedFile)
+    /// <param name="groupState">The state of the resulting <see cref="ImportGroup"/>.</param>
+    public static ImportGroup CompleteGroup(this IFileSource associatedFile, State groupState)
```
Comment (czdietrich) [inline src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs]: Shouldn't this be also called `CreateImportGroup`? The `Complete` seems to be misleading if the user can pass any `groupState` here.

Comment (czdietrich) [reply src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs]: We should ensure that the developer must pass the `primaryInputSource` to that group.

What do you think about changing `IImportFormat` interface slightly 
from `ImportGroup? BuildGroup(IFileSource primaryFileSource, IGroupContext context)` 
to `ImportGroup? BuildGroup(IPrimaryFileSource primaryFileSource, IGroupContext context)` 
and also add an interface like `IPrimaryFileSource : IFileSource`?

That way we could enforce the `primaryInputSource` to be part of that group.

File: src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs [hunk 2]
```diff
@@ -27,11 +27,12 @@ public static class ImportGroupExtensions
     /// Creates an <see cref="ImportGroup"/> that is complete and can be imported at once.
```
Comment (stephan-scholz-zeiss) [inline src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs:27]: The status no longer has to be complete.

File: src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs [hunk 3]
```diff
@@ -40,39 +41,12 @@ public static ImportGroup CreateCompleteImportGroup(this IFileSource associatedF
...
```
Comment (stephan-scholz-zeiss) [inline src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs:41]: The status no longer has to be complete.

**Review by stephan-scholz-zeiss — APPROVED** [review]

**Comment by czdietrich** [comment]
Some files seem to miss the copyright header. Could you please add it?

File: src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs (hunk 1 above)
Comment (czbarth) [reply src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs]: Good idea. I introduced `IPrimaryFileSource `and altered the code accordingly.

File: src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs (hunk 2 above)
Comment (czbarth) [reply src/PiWeb.Import.Sdk/Modules/ImportFormat/ImportGroupExtensions.cs]: I removed the extension completely because there was not much benefit in them.

**Pull request merged**